pkg/reconciler/autoscaling/kpa/controller.go

```diff
@@ -82,6 +85,18 @@ func NewController(
 	}
 	paInformer.Informer().AddEventHandler(paHandler)
 
+	// When we see PodAutoscalers deleted, clean up the decider.
+	paInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
```

Comment:
Should we model this as a finalizer to be more "correct" and consistent?

Reply:
Feels like overkill for the in-memory decider state.  The informer deletion event should be well suited to cover that since it also keys off of in-memory state.  It's somewhat tempting given your other comment, and trying to ensure coverage, but overkill. 😬